src/Nethermind/Nethermind.TxPool/Collections/BlobTxDistinctSortedPool.cs

```diff
@@ -53,6 +58,57 @@ public bool TryGetBlobAndProof(byte[] requestedBlobVersionedHash,
         return false;
     }
 
+    public bool TryGetBlobAndProofV2(byte[] requestedBlobVersionedHash,
+        [NotNullWhen(true)] out byte[]? blob,
+        [NotNullWhen(true)] out byte[][]? cellProofs)
+    {
+        using var lockRelease = Lock.Acquire();
+
+        if (BlobIndex.TryGetValue(requestedBlobVersionedHash, out List<Hash256>? txHashes))
+        {
+            foreach (Hash256 hash in CollectionsMarshal.AsSpan(txHashes))
+            {
+                if (TryGetValueNonLocked(hash, out Transaction? blobTx) && blobTx.BlobVersionedHashes?.Length > 0)
+                {
+                    for (int indexOfBlob = 0; indexOfBlob < blobTx.BlobVersionedHashes.Length; indexOfBlob++)
+                    {
+                        if (Bytes.AreEqual(blobTx.BlobVersionedHashes[indexOfBlob], requestedBlobVersionedHash)
+                            && blobTx.NetworkWrapper is ShardBlobNetworkWrapper wrapper)
+                        {
+                            if (wrapper is not { Version: ProofVersion.V1 })
+                            {
+                                break;
+                            }
+
+                            blob = wrapper.Blobs[indexOfBlob];
+                            cellProofs = wrapper.Proofs.Slice(Ckzg.Ckzg.CellsPerExtBlob * indexOfBlob, Ckzg.Ckzg.CellsPerExtBlob);
```

Comment:
Can individual cell proof be ReadOnlyMemory? We could do no-alloc then.

Reply:
[Added to todo](https://github.com/NethermindEth/nethermind/pull/8536)